homeassistant/components/tahoma/lock.py

```diff
@@ -0,0 +1,89 @@
+"""Support for Tahoma lock."""
+from datetime import timedelta
+import logging
+
+from homeassistant.components.lock import LockDevice
+from homeassistant.const import ATTR_BATTERY_LEVEL, STATE_LOCKED, STATE_UNLOCKED
+
+from . import DOMAIN as TAHOMA_DOMAIN, TahomaDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=120)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Tahoma lock."""
+    controller = hass.data[TAHOMA_DOMAIN]["controller"]
+    devices = []
+    for device in hass.data[TAHOMA_DOMAIN]["devices"]["lock"]:
+        devices.append(TahomaLock(device, controller))
+    add_entities(devices, True)
+
+
+class TahomaLock(TahomaDevice, LockDevice):
+    """Representation a Tahoma lock."""
+
+    def __init__(self, tahoma_device, controller):
+        """Initialize the device."""
+        super().__init__(tahoma_device, controller)
+        self._state = STATE_UNLOCKED
+        self._available = False
+        self._battery_level = None
+        self._name = None
+
+    def update(self):
+        """Update method."""
+        self.controller.get_states([self.tahoma_device])
+        self._battery_level = self.tahoma_device.active_states["core:BatteryState"]
+        self._name = self.tahoma_device.active_states["core:NameState"]
+        if self._battery_level == "low":
+            _LOGGER.warning("Lock %s has low battery", self._name)
+        if self._battery_level == "verylow":
+            _LOGGER.error("Lock %s has very low battery", self._name)
+        if self.tahoma_device.active_states.get("core:LockedUnlockedState") == "locked":
+            self._state = STATE_LOCKED
+        else:
+            self._state = STATE_UNLOCKED
+        self._available = (
+            self.tahoma_device.active_states.get("core:AvailabilityState")
+            == "available"
+        )
+
+    def open(self, **kwargs):
+        """Open method."""
+        pass
+
+    def unlock(self, **kwargs):
+        """Unlock method."""
+        _LOGGER.info("unlocking %s", self._name)
```

Comment:
Do not spam the logs with messages like this one.
```suggestion
        _LOGGER.debug("unlocking %s", self._name)
```